waku/incentivization/rpc_codec.nim

```diff
@@ -0,0 +1,63 @@
+import
+  std/options
+import
+  ../common/protobuf,
+  ../waku_core,
+  ./rpc
+
+
+# Codec for EligibilityProof
+
+proc encode*(epRpc: EligibilityProof): ProtoBuffer = 
+  var pb = initProtoBuffer()
+  if epRpc.proof.isSome:
+    let proof = epRpc.proof.get()
+    pb.write3(1, proof)
+  else:
+    # there is no proof
+    discard
+  pb
+
+proc decode*(T: type EligibilityProof, buffer: seq[byte]): ProtobufResult[T] = 
+  let pb = initProtoBuffer(buffer)
+  var epRpc = EligibilityProof()
+  var proof = newSeq[byte]()
+  if not ?pb.getField(1, proof):
+    epRpc.proof = none(seq[byte])
+  else:
+    epRpc.proof = some(proof)
+  ok(epRpc)
+
+
+# Codec for EligibilityStatus
+
+proc encode*(esRpc: EligibilityStatus): ProtoBuffer = 
+  var pb = initProtoBuffer()
+  if esRpc.statusCode.isSome:
+    pb.write3(1, esRpc.statusCode.get())
+  if esRpc.statusDesc.isSome:
+    pb.write3(2, esRpc.statusDesc.get())
+  pb
+
+proc decode*(T: type EligibilityStatus, buffer: seq[byte]): ProtobufResult[T] = 
+  let pb = initProtoBuffer(buffer)
+  var esRpc = EligibilityStatus()
+  # status code
+  # TODO: write this more concisely: ternary operator, default values?
+  # something like this, if hasField by field number existed:
+  #esRpc.statusCode = pb.hasField(1) ? some(pb.getField(1)) : none(uint32)
+  # the same applies to the EligibilityProof's decode
+  var code = uint32(0)
+  if not ?pb.getField(1, code):
```

Comment:
If there's a response I think there MUST be a status_code. In other words, raise an error here rather than just setting this to `none`. Generally we do consider the extra information in the `status_desc` to be optional.